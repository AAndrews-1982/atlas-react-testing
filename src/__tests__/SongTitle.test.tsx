// Test for SongTitle Component

// Imports
import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import { SongTitle } from "../components/SongTitle";
import { AppContext } from "../components/AppContext";

// Mock Context Provider Data
const mockContextValue = {
  songs: [
    { title: "Test Song", artist: "Test Artist", duration: "3:00", cover: "cover1.png" }
  ],
  currentSong: 0,
  setCurrentSong: () => {} // Mock function
};

test("SongTitle renders correctly", () => {
  const { asFragment } = render(
    <AppContext.Provider value={mockContextValue}>
      <SongTitle />
    </AppContext.Provider>
  );

  // Correctly call toMatchSnapshot() on asFragment
  expect(asFragment()).toMatchSnapshot();
});
