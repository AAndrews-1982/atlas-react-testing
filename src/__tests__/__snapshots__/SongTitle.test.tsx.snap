// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle renders correctly 1`] = `
<DocumentFragment>
  <div
    class="song-details"
  >
    <h2
      class="song-title mb-2 text-3xl font-bold"
    >
      Test Song
    </h2>
    <div
      class="artist-name mb-4 text-lg text-muted-text dark:text-dark-muted-text"
    >
      Test Artist
    </div>
  </div>
</DocumentFragment>
`;
